source/src/encryption/cs_ConnectionEncryption.cpp

```diff
@@ -144,6 +146,20 @@ void ConnectionEncryption::generateSessionData() {
 	RNG::fillBuffer(_sessionData.sessionNonce, sizeof(_sessionData.sessionNonce));
 	RNG::fillBuffer(_sessionData.validationKey, sizeof(_sessionData.validationKey));
 	memcpy(_nonce.sessionNonce, _sessionData.sessionNonce, sizeof(_sessionData.sessionNonce));
+	_log(LogLevelConnectionEncryption, false, "Set session data:");
+	_logArray(LogLevelConnectionEncryption, true, reinterpret_cast<uint8_t*>(&_sessionData), sizeof(_sessionData));
+}
+
+cs_ret_code_t ConnectionEncryption::setSessionData(session_data_t& sessionData) {
+	if (sessionData.protocol != _sessionData.protocol) {
+		return ERR_PROTOCOL_UNSUPPORTED;
+	}
+	memcpy(_sessionData.sessionNonce, sessionData.sessionNonce, sizeof(sessionData.sessionNonce));
+	memcpy(_sessionData.validationKey, sessionData.validationKey, sizeof(sessionData.validationKey));
```

Comment:
This can be replaced with direct assignment because default constructor will do a memberwise copy of fixed size arrays.
```suggestion
	_sessionData = sessionData;
```

*Note that protocol is checked for eq anyway, so no worries there.

Reply:
Hm, but right now it's similar to the code in generateSessionData().